src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -127,14 +129,21 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
         for (Task task : validTasks) {
             String chainTaskId = task.getChainTaskId();
 
-            // no need to ge further if the consensus is already reached on-chain
+            // Check if task is still in contribution phase with INITIALIZED or RUNNING status
+            Optional<Task> upToDateTask = taskService.getTaskByChainTaskId(chainTaskId);
+            if (upToDateTask.isEmpty() || ! TaskStatus.isInContributionPhase(upToDateTask.get().getCurrentStatus())) {
```

Comment:
I see your point but that patch looks a bit strange to me.
The design of this method should be updated.

In the mean time, in addition to moving most safety checks outside this for loop, I would move your patch right before `replicatesService.addNewReplicate(chainTaskId, walletAddress);`, in order to decrease chances of giving a task/replicate with an outdated status.

Reply:
We need to discuss on how this method should be updated then to specify and implement things correctly.

I don't know if we want to do it right now or send this first patch and rework this implementation early January.

Reply:
Check moved to a later part of the code, just before retrieving replicates and computing if more contributions are required
See 99555cb